File: app/src/main/java/pl/pola_app/model/Report.java
package pl.pola_app.model;

import com.google.gson.annotations.SerializedName;

/**
 * Created by Tajchert on 28.10.2015.
 */
public class Report {
    @SerializedName("description")
    String description;
    @SerializedName("product_id")
    String productId;
    @SerializedName("files_count")
    int filesCount;
    @SerializedName("mime_type")
    String mimeType;
    @SerializedName("file_ext")
    String fileExt;

    public Report(String description, String productId, int filesCount, String mimeType, String fileExt) {
        this(description, filesCount, mimeType, fileExt);
        this.productId = productId;
    }

    public Report(String description, int filesCount, String mimeType, String fileExt) {
        this.description = description;
        this.filesCount = filesCount;
        this.mimeType = mimeType;
        this.fileExt = fileExt;
    }
}

File: app/src/main/java/pl/pola_app/model/Report.kt
package pl.pola_app.model

import com.google.gson.annotations.SerializedName

/**
 * Created by Tajchert on 28.10.2015.
 */
data class Report(
    @field:SerializedName("description") val description: String,
    @field:SerializedName("files_count") val filesCount: Int,
    @field:SerializedName("mime_type") val mimeType: String,
    @field:SerializedName("file_ext") val fileExt: String
) {
    @SerializedName("product_id")
    var productId: String = ""

    constructor(
        description: String,
        productId: String,
        filesCount: Int,
        mimeType: String,
        fileExt: String
    ) : this(description, filesCount, mimeType, fileExt) {
        this.productId = productId
    }
}